chart/kubeapps/templates/kubeapps-frontend-config.yaml

```diff
@@ -57,36 +57,42 @@ data:
       # Ensure each cluster can be reached (should only be
       # used with an auth-proxy where k8s credentials never leave
       # the cluster). See clusters option.
-      {{- range .Values.clusters }}
+    {{- range .Values.clusters }}
       location ~* /api/clusters/{{ .name }} {
-        rewrite /api/clusters/{{ .name }}/(.*) /$1 break;
-        rewrite /api/clusters/{{ .name }} / break;
-
-      {{/* Helm returns a nil pointer error when accessing foo.bar if foo doesn't
+    {{/* We need to split the API service URL(s) into the base url and the path segment so
+           those configurations using a path can be appropriately rewritten below while
+           ensuring the proxy_pass statement is given the base URL only. */}}
+    {{- $parsed := urlParse (default "https://kubernetes.default" .apiServiceURL) }}
+    {{- $apiServiceBaseURL := urlJoin (pick $parsed "scheme" "host") }}
+    {{- $apiServiceURLPath := $parsed.path }}
+        rewrite /api/clusters/{{ .name }}/(.*) {{ $apiServiceURLPath }}/$1 break;
+        rewrite /api/clusters/{{ .name }} {{ $apiServiceURLPath }}/ break;
+
+    {{/* Helm returns a nil pointer error when accessing foo.bar if foo doesn't
            exist, even with the `default` function.
            See https://github.com/helm/helm/issues/8026#issuecomment-756538254 */}}
-      {{- $pinnipedConfig := .pinnipedConfig | default dict }}
-      {{- if and $.Values.pinnipedProxy.enabled $pinnipedConfig.enable }}
+    {{- $pinnipedConfig := .pinnipedConfig | default dict }}
+    {{- if and $.Values.pinnipedProxy.enabled $pinnipedConfig.enable }}
         # If pinniped proxy is enabled *and* the current cluster is configured
         # to exchange credentials then we route via pinnipedProxy to exchange
         # credentials for client certificates.
-        {{- if .apiServiceURL }}
+    {{- if .apiServiceURL }}
         proxy_set_header PINNIPED_PROXY_API_SERVER_URL {{ .apiServiceURL }};
-        {{- end }}
-        {{- if .certificateAuthorityData }}
+    {{- end }}
+    {{- if .certificateAuthorityData }}
         proxy_set_header PINNIPED_PROXY_API_SERVER_CERT {{ .certificateAuthorityData }};
-        {{- end }}
+    {{- end }}
         proxy_pass http://kubeapps-internal-pinniped-proxy.{{ $.Release.Namespace }}:{{ $.Values.pinnipedProxy.service.port }};
-      {{- else }}
+    {{- else }}
         # Otherwise we route directly through to the clusters with existing credentials.
-        proxy_pass {{ default "https://kubernetes.default" .apiServiceURL }};
-        {{- if .certificateAuthorityData }}
+        proxy_pass {{ $apiServiceBaseURL }};
```

Comment:
No `default` required here now that it happens above one line 65.

Reply:
Yep, see above.